src/functions/onLeaveGame.ts

```diff
@@ -0,0 +1,75 @@
+import { Handler } from 'aws-lambda';
+import { deleteGame, removeUserFromGame } from '../module/gameDBService';
+import {
+  GameStates,
+  LambdaEventBody,
+  LambdaEventBodyPayloadOptions,
+  LambdaResponse,
+  WebSocketActions,
+  WebSocketAPIGatewayEvent,
+} from '../types';
+import { response } from '../utils/response';
+import { Logger } from '../utils/Logger';
+import { WebSocketClient } from '../WebSocketClient';
+import { createLeaveResponse, failedWebSocketResponse, successWebSocketResponse } from '../utils/webSocketActions';
+import { broadcastGameUpdate, broadcastInGameMessage, broadcastInGameUpdate } from '../utils/broadcast';
+import { Game } from '../models/Game';
+import { removeGameDocumentVersion } from '../utils/dbHelper';
+
+/**
+ * Handler for joining a game.
+ * @param {WebSocketAPIGatewayEvent} event Websocket API gateway event
+ */
+export const handler: Handler = async (event: WebSocketAPIGatewayEvent): Promise<LambdaResponse> => {
+  // Logger
+  Logger.createLogTitle('onJoinGame.ts');
+
+  // Parse event
+  const { connectionId } = event.requestContext;
+  const body: LambdaEventBody = JSON.parse(event.body);
+  const { payload }: { payload: LambdaEventBodyPayloadOptions } = body;
+  const gameId = payload.gameId as string;
+
+  // Leave game
+  console.log('Leaving a game...');
+  const ws = new WebSocketClient(event.requestContext);
+  try {
+    // Remove user from game
+    const updatedGame = (await removeUserFromGame(gameId, connectionId)) as Game;
+    removeGameDocumentVersion<Game>(updatedGame);
+    console.log('Updated game after leaving a game:', updatedGame);
+
+    // Send success response
+    const res = createLeaveResponse(updatedGame);
+    const updatedGameResponse = successWebSocketResponse(res);
+    await ws.send(JSON.stringify(updatedGameResponse), connectionId);
+
+    // Send message to other users in the game
+    // 1. If the host leaves the game,
+    // send GAME_UPDATE with DELETE state to other user in the game and also delete the game in the table
+    const { host } = updatedGame;
+    if (host.connectionId === connectionId) {
+      await broadcastGameUpdate(ws, gameId, GameStates.DELETED, connectionId);
+      await deleteGame(gameId);
+    } else {
```

Comment:
Other connections is missing a LEAVE_GAME message if host leaves. Check diagram. Unless you think client should check whether to leave their game via GAME_UPDATE if it has been deleted. Otherwise, I think adding ```      const connectionIds = updatedGame.users.map((user) => user.connectionId);
      await broadcastInGameMessage(ws, connectionId, WebSocketActions.LEAVE_GAME, connectionIds);``` is sufficient

Reply:
Ignore above code. Send a leave response to the other connectionIds with the already generated ```updatedGameResponse```

Reply:
ideally, when the host left the game, a random user should be switched to be the host and can be in charge of starting a game or not

Reply:
put I can put in that fix for now based on the design

Reply:
```ideally, when the host left the game, a random user should be switched to be the host and can be in charge of starting a game or not```
This approach is ok. We can leave this for next PR